araproc/analysis/standard_reco.py

```diff
@@ -924,3 +926,346 @@ def get_avg_corr_snr(self, wave_packet, excluded_channels = []):
         avg_hpol_corr_snr = np.mean(avg_hpol_corr_snr)
 
         return avg_vpol_corr_snr, avg_hpol_corr_snr  
+
+    def __get_arrival_delays_reco(
+        self, data, reco_results, channels_to_csw, reference_ch, 
+        which_distance, solution
+    ):
+        """
+        Extract the arrival time of the signals from the reconstruction then 
+        calculate the time delays between each channel and the reference channel
+
+        Parameters
+        ----------
+        data : AnalysisDataset
+        reco_results : dict
+            Reco results already with the specific reconstruction reqeusted (so
+            the keys of this object include 'theta' and 'phi')
+        channels_to_csw : list
+            A list of channels IDs to calculate arrival delays for
+        reference_ch : int
+            Reference channel for use calculating arrival delays
+        which_distance : str
+            The distance of the reconstruction for analysis, following the 
+            convention of other functions in this class. 
+        solution : int
+            0 for direct ray solution. 1 for reflected/refracted ray solution.
+
+        Returns
+        -------
+        arrival_delays : dict
+            Dictionary where keys are channel numbers and values are arrival 
+            delays calculated based on expected arrival times from the 
+            reconstructed event vertex.
+        """
+
+        # Get all the arrival times calculated based on the reconstructed
+        #   event vertex and the time tables
+        arrival_times = {ch: 0.0 for ch in channels_to_csw }
+        for c, ch_ID in enumerate(channels_to_csw):
+            arrival_times[ch_ID] = self.lookup_arrival_time(
+                ch_ID, reco_results['theta'], reco_results['phi'], 
+                which_distance=which_distance, solution=solution
+            )
+        
+        # Calculate time difference between each channel's arrival time
+        #   and the reference channel's arrival time
+        reference_arrival_time = arrival_times[reference_ch]
+        arrival_delays = {ch: 0.0 for ch in channels_to_csw }
+        for ch_ID in arrival_delays.keys():
+            arrival_delays[ch_ID] = arrival_times[ch_ID] - reference_arrival_time
+
+        return arrival_delays
+
+    def __get_arrival_delays_AraRoot_xcorr(
+        self, wavepacket, pol, channels_to_csw, reference_ch, reco_delays, 
```

Comment:
Looks like `pol` isn't used here. Let's drop it.

Reply:
Yep thank you!